### PR TITLE
Cleanup ForwardWalking class

### PR DESCRIPTION
**Pull request by PhilipFackler**
## Proposed changes

Renamed and made private all data members
Refactored functions from h to cpp
Removed unused code
Related to #3411, #4424

## What type(s) of changes does this code introduce?

- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ubuntu 22.04

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCHamiltonians/ForwardWalking.cpp [hunk 1]
```diff
@@ -24,18 +24,66 @@
 
 namespace qmcplusplus
 {
+ForwardWalking::ForwardWalking() { update_mode_.set(OPTIMIZABLE, 1); }
+
+ForwardWalking::~ForwardWalking() {}
```
Comment (ye-luo) [inline src/QMCHamiltonians/ForwardWalking.cpp:29]: `ForwardWalking::~ForwardWalking() = default;`

File: src/QMCHamiltonians/ForwardWalking.cpp [hunk 2]
```diff
@@ -24,18 +24,66 @@
...
+
+std::string ForwardWalking::getClassName() const { return "ForwardWalking"; }
```
Comment (ye-luo) [inline src/QMCHamiltonians/ForwardWalking.cpp:31]: Please keep this defined in the header file.

File: src/QMCHamiltonians/ForwardWalking.cpp (hunk 1 above)
Comment (PhilipFackler) [reply src/QMCHamiltonians/ForwardWalking.cpp]: done

File: src/QMCHamiltonians/ForwardWalking.cpp (hunk 2 above)
Comment (PhilipFackler) [reply src/QMCHamiltonians/ForwardWalking.cpp]: done

File: src/QMCHamiltonians/ForwardWalking.h [hunk 3]
```diff
@@ -18,99 +18,57 @@
...
-
+public:
   /** constructor
    */
-  ForwardWalking() { update_mode_.set(OPTIMIZABLE, 1); }
+  ForwardWalking();
 
   ///destructor
-  ~ForwardWalking() override {}
+  ~ForwardWalking() override = default;
```
Comment (ye-luo) [inline src/QMCHamiltonians/ForwardWalking.h:34]: Actually I found it beneficial to have the desctructor defined in cpp.
Leave
```
~ForwardWalking() override;
```
in the header file
and put
```
~ForwardWalking() override = default;
```
in the cpp.
When unique_ptr is used, this separation enables forward declaration.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**